spec/game_spec.rb

```diff
@@ -6,44 +6,52 @@
 require 'board'
 require 'player'
 require 'number_validator'
+require 'outcome_checker'
+require 'ui'
 require 'stringio'
 require 'pry'
 
-class TestConsole
-  attr_reader :messages
+class TestUIForGame
+  attr_accessor :output
 
   def initialize
-    @messages = []
+    @output = []
   end
 
-  def output(message)
-    @messages << message
+  def welcome
+    output << 'Welcome to Tic Tac Toe!'
+  end
+
+  def goodbye
+    output << 'Thank you for playing. Goodbye!'
   end
 end
 
 class TestGameLooper
-  attr_reader :console
+  attr_accessor :methods_called
 
-  def initialize(console)
-    @console = console
+  def initialize
+    @methods_called = []
   end
 
   def loop
-    console.output('Looping')
+    methods_called << 'Loop'
```

Comment:
**suggestion (non-blocker)**: Rename this to `messages_sent`.

I think `messages_sent` might help you think more in terms of "messages sent" vs "methods called".